### PR TITLE
[FINE] Disable connection reuse for WebSocket connections in Apache

### PR DESCRIPTION
**Pull request by skateman**
This is a temporary workaround for the issue described here:
https://bugzilla.redhat.com/show_bug.cgi?id=1404354

This can be reverted after httpd is updated to 2.4.25 or newer

### CODE REVIEW

File: lib/gems/pending/util/miq_apache/miq_apache.rb [hunk 1]
```diff
@@ -266,15 +266,21 @@ def add_ports(ports, protocol)
...
+        end
       end
       ports
     end
 
     def remove_ports(ports, protocol)
       ports = Array(ports)
       ports.each do |port|
-        @raw_lines.delete_if { |line| line =~ /BalancerMember\s+#{protocol}:\/\/0\.0\.0\.0:#{port}$/ }
+        @raw_lines.delete_if { |line| line =~ /BalancerMember\s+#{protocol}:\/\/0\.0\.0\.0:#{port}( disablereuse=on)?$/ }
```
Comment (jrafanie) [inline lib/gems/pending/util/miq_apache/miq_apache.rb]: @skateman does this mean we only remove the `ws` protocol BalancerMembers?  What about non-ws?  I guess I'm wondering why we are checking for disablereuse at all if no other line will match the protocol, 0.0.0.0 and port... Right?

Comment (skateman) [inline lib/gems/pending/util/miq_apache/miq_apache.rb]: Well, AFAIK the `?` at the end of the regexp makes the `disablereuse` part optional, but maybe I'm wrong.

Comment (jrafanie) [inline lib/gems/pending/util/miq_apache/miq_apache.rb]: Yeah, I believe so, but if it's not required, I wouldn't include it. I think it's not required so I'd prefer we keep the regexp as simple as possible.  Maybe I'm wrong.

Comment (skateman) [inline lib/gems/pending/util/miq_apache/miq_apache.rb]: Well, I saw the `$` at the end and did not wanted to drop it.

File: spec/util/miq_apache/conf_spec.rb [hunk 2]
```diff
@@ -88,8 +88,22 @@
...
+    context "with websocket as protocol" do
+      it 'add_ports should add the disablereuse=on suffix' do
         @conf.add_ports(3000, "ws")
-        expect(@conf.raw_lines).to eq(["<Proxy balancer://evmcluster/ lbmethod=byrequests>\n", "BalancerMember ws://0.0.0.0:3000\n", "</Proxy>\n"])
+        expect(@conf.raw_lines).to eq(["<Proxy balancer://evmcluster/ lbmethod=byrequests>\n", "BalancerMember ws://0.0.0.0:3000 disablereuse=on\n", "</Proxy>\n"])
+      end
+
+      it 'remove_ports should remove the line with a suffix' do
+        before = @conf.raw_lines.dup
+        @conf.raw_lines = ["<Proxy balancer://evmcluster/ lbmethod=byrequests>\n", "BalancerMember ws://0.0.0.0:3000 disablereuse=on\n", "</Proxy>\n"]
```
Comment (jrafanie) [inline spec/util/miq_apache/conf_spec.rb]: @skateman can you add a non-ws remove_ports test to confirm the regexp works for both?

Comment (skateman) [inline spec/util/miq_apache/conf_spec.rb]: @jrafanie done

**Comment by simaishi** [comment]
@skateman this can be `euwe/yes`?

**Comment by skateman** [comment]
We aren't using websocket notifications in `euwe`, but backporting this would definitely not do any harm.

**Comment by simaishi** [comment]
Then... what change will fix the issue for Euwe branch??

**Comment by skateman** [comment]
@simaishi uh, I got confused by the names, sure, backport it to `euwe`, all the things I said was ment for `darga` :wink: sorry

**Comment by simaishi** [comment]
@skateman Thanks 😄 

Euwe backport (to manageiq repo) details:

```
$ git log -1
commit eee139331f831587ee62d3979c074eb201520c84
Author: Satoe Imaishi <simaishi@redhat.com>
Date:   Thu Jul 6 09:56:11 2017 -0400

    Merge pull request #219 from skateman/fine-websocket-fix
    
    [FINE] Disable connection reuse for WebSocket connections in Apache
    (cherry picked from commit 73a5739b13fc460b4a339ea80571111940dbb808)
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1468633
```

**Pull request merged**